src/Wallet.purs

```diff
@@ -167,7 +170,100 @@ dummySign tx@(Transaction { witnessSet: tws@(TransactionWitnessSet ws) }) =
     )
 
 -------------------------------------------------------------------------------
--- FFI stuff
+-- Gero backend
+-------------------------------------------------------------------------------
+-- Mirror of NamiWallet backend for Gero
+type GeroWallet =
```

Comment:
This is identical to `NamiWallet`. I imagined that the two would be similar, but I didn't foresee that they would be exactly the same. My suggestion would be the following:

- define a type `Cip30Wallet`:
  ```purescript
  type Cip30Wallet (a :: Type) =
    { connection :: Ref.Ref a
    , getWalletAddress :: a -> Aff (Maybe Address)
    , getCollateral :: a -> Aff (Maybe TransactionUnspentOutput)
    , signTx :: a -> Transaction -> Aff (Maybe Transaction)
    }
  ```
- turn `NamiWallet` into `Cpi30Wallet NamiConnection`, similar for `GeroWallet`
- we probably want to de-duplicate `mkNamiWalletAff` and `mkGeroWalletAff`, provided their behavior really is the same (see my other comment below about datums)
- we could also simplify the FFI stuff in this case, e.g. by passing the name of the API object into a set of de-duplicated FFI functions

We can also remove `submitTx` from both wallets, as we now use Ogmios to submit

This would make maintenance a lot easier IMO

Reply:
Removed `NamiConnection` and `GeroConnection` altogether in favor of `Cip30Connection`; we don't need custom logic for submission since we use Ogmios now.